### PR TITLE
MM-T1787 Toast does not appear when all new messages are visible without scrolling down

### PR DESCRIPTION
**Pull request by openmohan**
#### Summary

MM-T1787 Toast does not appear when all new messages are visible without scrolling down

#### Ticket Link

[Usecase](https://automation-test-cases.vercel.app/test/MM-T1787)

### CODE REVIEW

**Comment by openmohan** [comment]
> Thanks @openmohan, looks good so far. However, it seemed to fall short on the next test step no. 2 which is:
> 
> ```
> Steps #2
> 1. Scroll up
> 
> Expected
> - No toast appears as all content below has already been seen
> ```

My bad :-).
I have added `scroll up` scenario too. Please review when you find the time.

**Comment by openmohan** [comment]
> Thanks for the update and no worries on the other scenario :).
> 
> I'm all good except for last minor comments that I missed during initial review.
> 
> Also, in the future, please avoid force pushing. That will be difficult for the reviewers to follow along with the changes as we progress on PR feedback/resolution. Just add the changes and then push to your branch. We have blog related to PR and this was mentioned on item 4 - https://developers.mattermost.com/blog/submitting-great-prs/#4-avoid-force-pushing

Thanks for the suggestion. I will avoid force pushing. I wanted to keep in a single commit that is why I was force pushing. But yeah, I should have done it once the reviews are done or Is it ok to have multiple commits for a feature/test? :-)

File: e2e/cypress/integration/toast/toast_spec.js [hunk 1]
```diff
@@ -263,6 +265,39 @@ describe('toasts', () => {
             cy.get('div.toast').should('be.visible').contains('Viewing message history');
         });
     });
+
+    it.only('MM-T1787 Toast does not appear when all new messages are visible without scrolling down', () => {
```
Comment (saturninoabril) [inline e2e/cypress/integration/toast/toast_spec.js]:
```suggestion
    it('MM-T1787 Toast does not appear when all new messages are visible without scrolling down', () => {
```

**Comment by saturninoabril** [comment]
It's okay to have single or multiple commits before PR submission depending on how you wanted to organize it. Once the review has started, it would be nice to push a single commit only though multiple commits are fine too if there are too many changes or grouping by comments makes sense. The only things to watch out are rebasing and force pushing.

File: e2e/cypress/integration/toast/toast_spec.js [hunk 2]
```diff
@@ -263,6 +265,39 @@ describe('toasts', () => {
...
+
+        // # Visit town square channel
+        visitTownSquareAndWaitForPageToLoad();
+
+        // * Assert toast should not be present as the messages are visible without scrolling down
+        cy.get('div.toast').should('not.be.visible');
+
+        // # Move to the top of the channel
+        cy.get('div.post-list__dynamic').should('be.visible').scrollTo('top', {duration: TIMEOUTS.ONE_SEC});
+
```
Comment (josephbaylon) [inline e2e/cypress/integration/toast/toast_spec.js:297]: I would suggest to add verification that `channelIntro` is visible to make sure it is scrolled to the top.

**Comment by openmohan** [comment]
Working on fixing the commits

**Comment by openmohan** [comment]
> LGTM. Just one minor verification comment. Thanks @openmohan !!

Thanks for the review. I have added the suggestion.

File: e2e/cypress/integration/toast/toast_spec.js [hunk 3]
```diff
@@ -263,6 +265,42 @@ describe('toasts', () => {
...
+        // * Verify that off-topic channel is loaded
```
Comment (josephbaylon) [inline e2e/cypress/integration/toast/toast_spec.js:298]: Nit-pick: Update comment
```suggestion
        // * Verify that town-square channel is loaded
```

**Comment by openmohan** [comment]
> LGTM. Thanks @openmohan !!

Thanks for reviewing and adding the suggestions :+1:  @josephbaylon

**Pull request merged**